File: java/src/main/java/com/example/fileservice/config/MinioConfiguration.java
package com.example.fileservice.config;

import io.minio.MinioClient;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@RequiredArgsConstructor
public class MinioConfiguration {


    private final MinioProperties minioProperties;

    @Bean
    public MinioClient minioClient() {
        return
                MinioClient.builder()
                        .endpoint(minioProperties.getHost())
                        .credentials(minioProperties.getAccessKey(), minioProperties.getSecretKey())
                        .build();
    }
}

File: kotlin/src/main/kotlin/com/example/fileservice/config/MinioConfiguration.kt
package com.example.fileservice.config

import io.minio.MinioClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class MinioConfiguration(val minioProperties: MinioProperties) {
    @Bean
    fun minioClient(): MinioClient {
        return MinioClient.builder()
            .endpoint(minioProperties.host)
            .credentials(minioProperties.accessKey, minioProperties.secretKey)
            .build()
    }
}
